ts/components/services/ContactPreferencesToggles/PreferenceToggleRow.tsx

```diff
@@ -24,15 +32,49 @@ const PreferenceToggleRow = ({
   label,
   onPress,
   value,
+  isLoading,
+  isError,
+  onReload,
+  disabled,
   testID = "preference-toggle-row"
 }: Props): React.ReactElement => (
   <View style={[styles.row]}>
     <View style={IOStyles.flex}>
-      <H4 weight={"Regular"} color={"bluegreyDark"}>
+      <H4
+        weight={"Regular"}
+        color={"bluegreyDark"}
+        testID={`${testID}-label`}
+        accessibilityRole={"text"}
+      >
         {label}
       </H4>
     </View>
-    <Switch value={value} onValueChange={onPress} testID={testID} />
+    {!isLoading && !isError && (
```

Comment:
good point, I was actually thinking of a possible alternative solution since I didn't like that much a group of boolean conditions. Going to implement this refactoring and fix tests too.